src/niclips/image/_convert.py

```diff
@@ -1,21 +1,18 @@
-import warnings
-
 import matplotlib as mpl
 import nibabel as nib
 import numpy as np
-from nilearn.image import reorder_img, resample_img
+from numpy.typing import ArrayLike
 from PIL import Image
 
 from niclips.typing import NiftiLike
 
 EPS = 1e-8
 
 
-def get_fdata(img: NiftiLike) -> np.ndarray:
+def get_fdata(img: NiftiLike) -> ArrayLike:
```

Comment:
Hm, this function no longer does what it claims to do. If I understand correct, `dataobj` may be either in memory or be a memmap, and can have arbitrary data type. It feels somewhat risky to me to interact directly with the `dataobj`. Also especially if we are now using the `nifti` lib, which always returns an in-memory ndarray.

Reply:
I've reverted the conditional back to using the `get_fdata()` in 9564b03 and tested to make sure there is no direct interaction with the `dataobj`. More of a comment for my own sake - the array **may** get resampled downstream so don't want to manipulate the original data loaded in.

An alternative is just calling `np.asarray(img.dataobj, copy=True)`, but I think the concern with the arbitrary data type will still be there. From what I can tell, the implementation in the current version of `nibabel` we are using is always returning a copy if using the `np.asarray` method anyways.